modules/common/src/DataResource.php

```
@@ -344,7 +394,12 @@ private static function getDistribution($identifier) {
    * Generates MD5 checksum for a file.
    */
   public function generateChecksum() {
-    $this->checksum = md5_file($this->filePath);
```

Can this get run on other perspectives? Is it possible that failing silently like this would have unexpected effects? Consider adding try/catch